test/utils/ReceiptDataHelperSpec.scala

```diff
@@ -0,0 +1,143 @@
+    1, 2, 3, 4, 5, 6, 7, 8, 9, flatRateScheme, dateToUse, dateToUse, dateToUse
+  )
+
+  def expectedAnswers(frs: Boolean): Seq[Answers] = {
+    val boxSixExpected: String = if (frs) "Total value of sales and other supplies, including VAT" else
+      "Total value of sales and other supplies, excluding VAT"
+    Seq(Answers(
+      "Nine box submission",
+      Seq(
+        ("box1", "VAT you charged on sales and other supplies", 1),
+        ("box2", "VAT you owe on goods purchased from EC countries and brought into the UK", 2),
+        ("box3", "VAT you owe before deductions (this is the total of box 1 and 2)", 3),
+        ("box4", "VAT you have claimed back", 4),
+        ("box5", "Return total", 5),
+        ("box6", boxSixExpected, 6),
+        ("box7", "Total value of purchases and other expenses, excluding VAT", 7),
+        ("box8", "Total value of supplied goods to EC countries and related costs (excluding VAT)", 8),
+        ("box9", "Total value of goods purchased from EC countries and brought into the UK, as well as any related costs (excluding VAT)", 9)
+      ).map { case (questionId, question, answer) =>
+        Answer(questionId, question, Some(answer.toString))
+      }))
+  }
+
+  def expectedDeclaration(isAgent: Boolean): Declaration = {
+    Declaration(
+      if (isAgent) "I confirm that my client has received a copy of the information contained in this return and approved the information as being correct and complete to the best of their knowledge and belief."
+      else "By submitting this return, you are making a legal declaration that the information is correct and complete to the best of your knowledge and belief. A false declaration can result in prosecution.",
+      "Scarlett Flamberg", None, declarationConsent = true
+    )
+  }
+
+  private def mockOutboundCall(frs: Boolean, noName: Boolean = false) = {
+    (mockConnector.getCustomerDetails(_: String)(_: HeaderCarrier, _: ExecutionContext))
+      .expects(*, *, *)
+      .returning(Future.successful(Right(CustomerDetails(
+        if(noName) None else Some("Scarlett"), if(noName) None else Some("Flamberg"), None, None, frs
+      ))))
+  }
+
+  private def mockFailedOutboundCall() = {
+    (mockConnector.getCustomerDetails(_: String)(_: HeaderCarrier, _: ExecutionContext))
+      .expects(*, *, *)
+      .returning(Future.successful(Left(BadRequestError("UH_OH", "WE'RE IN TROUBLE"))))
+  }
+
+  def userWithCookie(cookieString: String): User[AnyContentAsEmpty.type] = User[AnyContentAsEmpty.type](vrn)(fakeRequest
+    .withCookies(Cookie(
+      "PLAY_LANG", cookieString, None, "/", None, secure = false, httpOnly = true
+    )))
+
+  def agentUserWithCookie(cookieString: String): User[AnyContentAsEmpty.type] = User[AnyContentAsEmpty.type](vrn, Some(arn))(fakeRequestWithClientsVRN
+    .withCookies(Cookie(
+      "PLAY_LANG", cookieString, None, "/", None, secure = false, httpOnly = true
+    )))
+
+  "extractReceiptData" should {
```

Comment:
I'm not sure on writing code and loops to generate scenarios for tests. You run the risks of this code being wrong and missing something, as well as running every combination of FRS, agent status and language values - its not really necessary when the combinations of values don't always have an effect on each other
The data in the model isn't huge, I don't see the problem in explicitly asserting each expected box value with the actual value, and for any extra scenarios only testing the bit that is different (similar to how we don't retest the entire view when testing for agent content). It also makes it more readable and easier to debug